src/integration-test/java/com/mongodb/jdbc/integration/testharness/IntegrationTestUtils.java

```diff
@@ -0,0 +1,429 @@
+            for (int i = 0; i < columnCount; i++) {
+                assertEquals(test.expected_is_writable.get(i), rsMetaData.isWritable(i + 1));
+            }
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private static void validateResultsOrdered(TestEntry testEntry, ResultSet rs)
+            throws SQLException {
+        Integer actualRowCounter = null;
+        if (testEntry.expected_result != null) {
+            while (rs.next()) {
+                if (actualRowCounter == null) {
+                    actualRowCounter = 0;
+                }
+                assertTrue(
+                        compareRow(
+                                (List<Object>) testEntry.expected_result.get(actualRowCounter),
+                                rs));
+                actualRowCounter++;
+            }
+        }
+        if (testEntry.row_count != null) {
+            validateRowCount(testEntry.row_count, testEntry, actualRowCounter, rs);
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private static void validateResultsUnordered(TestEntry testEntry, ResultSet rs)
+            throws SQLException {
+        Integer actualRowCounter = null;
+        if (testEntry.expected_result != null) {
+            while (rs.next()) {
+                boolean found = false;
+                if (actualRowCounter == null) {
+                    actualRowCounter = 0;
+                }
+                actualRowCounter++;
+                for (Object expectedRow : testEntry.expected_result) {
+                    if (compareRow((List<Object>) expectedRow, rs)) {
+                        found = true;
+                        break;
+                    }
+                }
+                assertTrue(found);
+            }
+        }
+        if (testEntry.row_count != null) {
+            validateRowCount(testEntry.row_count, testEntry, actualRowCounter, rs);
+        }
+    }
+
+    private static void validateRowCount(
+            int expectedRowCount, TestEntry testEntry, Integer actualRowCounter, ResultSet rs)
+            throws SQLException {
+        if (actualRowCounter == null) {
+            actualRowCounter = MongoIntegrationTest.countRows(rs);
+        }
+        if (testEntry.row_count_gte != null && testEntry.row_count_gte) {
+            assertTrue(actualRowCounter >= expectedRowCount);
+        } else {
+            assertEquals(java.util.Optional.ofNullable(actualRowCounter), expectedRowCount);
+        }
+    }
+
+    private static boolean compareRow(List<Object> expectedRow, ResultSet actualRow)
+            throws SQLException {
+        ResultSetMetaData rsMetadata = actualRow.getMetaData();
+        assertEquals(expectedRow.size(), rsMetadata.getColumnCount());
+
+        for (int i = 0; i < expectedRow.size(); i++) {
+            // Handle expected field being null
+            if (expectedRow.get(i) == null) {
+                if (actualRow.getObject(i + 1) == null) {
+                    break;
```

Comment:
Why do we break here? Doesn't this mean there is a null at position `i` in the expected result but that's the only place it is expected? As in, `expected_result: [1, 2, null, 3]` why would we want to break on the third value? Or am I misunderstanding what's going on here?

Reply:
That's a good catch.  It's supposed to be `continue` not `break`.  I must have gotten mixed up with seeing the `break`s in the switch statement.